source/postcard-rpc/src/server/mod.rs

```diff
@@ -425,6 +431,8 @@ where
                 buf,
                 dis: d,
             } = self;
+            rx.wait_connection().await;
```

Comment:
Might be nice to join these two futures together, but not a big deal practically either way.

Reply:
I've decided not to. Mostly because it doesn't really matter as i'd expect most of these would just wait for a state change (where order doesn't matter) rather then doing actual work. Also doing a join would require a dependency on e.g. futures-utils in the runtime-independent code, which doesn't really seem worth it for this.

Fwiw this made me wonder if this isn't hinting that a better abstraction would be one stream with a readhalf and write half (like {Unix,Tcp}Stream) in tokio. Where the connection waiting is on the stream rather then on the halves. (As i'd expect all postcard wire transports would be bidrectional).